include/deal.II/numerics/data_out.h

```diff
@@ -316,42 +313,117 @@ class DataOut : public DataOut_DoFData<DoFHandlerType,
                 const unsigned int     n_subdivisions = 0,
                 const CurvedCellRegion curved_region  = curved_boundary);
 
+  /**
+   * A function that allows selecting for which cells output should be
+   * generated. This function takes two arguments, both `std::function`
+   * objects that can be used what the first cell on which output is
+   * generated is supposed to be, and what given one cell the next
+   * function is supposed to be. Through these function objects,
+   * it is possible to select a subset of cells on which output should
+   * be produced (e.g., only selecting those cells that belong to a
+   * part of the domain -- say, the fluid domain in a code such as step-46),
+   * or to completely change *where* output is produced (e.g., to produce
+   * output on non-active cells of a multigrid hierarchy or if the finest
+   * level of a mesh is so fine that generating graphical output would lead
+   * to an overwhelming amount of data.
+   *
+   * @param[in] A function object that takes as argument the triangulation
+   *   this class works on and that should return the first cell on which
+   *   output should be generated.
+   * @param[in] A function object that takes as argument the last cell
+   *   on which output was generated and that should return the next
+   *   cell on which output should be generated. (In practice, the
+   *   function object may be triggered *before* any processing
+   *   actually happens, but you get the idea.)
```

Comment:
I documented both. Since the triangulation is so easy to obtain, I think there's no point in passing it explicitly (and most implementations wouldn't need it because they just do something like `++cell`.

Reply:
Actually, I change my mind.